lib/logging/GhostLogger.js

```diff
@@ -297,34 +297,30 @@ class GhostLogger {
      * logging.info({}, {}) --> is one object
      * logging.error(new Error()) --> is {err: new Error()}
      */
-    log(type, args) {
-        let modifiedArguments;
-
-        each(args, function (value) {
-            if (value instanceof Error) {
-                if (!modifiedArguments) {
-                    modifiedArguments = {};
-                }
-
-                modifiedArguments.err = value;
-            } else if (isObject(value)) {
-                if (!modifiedArguments) {
-                    modifiedArguments = {};
-                }
-
-                var keys = Object.keys(value);
-                each(keys, function (key) {
-                    modifiedArguments[key] = value[key];
-                });
-            } else {
-                if (!modifiedArguments) {
-                    modifiedArguments = '';
-                }
-
-                modifiedArguments += value;
-                modifiedArguments += ' ';
-            }
-        });
+     log(type, args) {
+         let modifiedMessages = [];
+         let modifiedObject = {};
+         let modifiedArguments = [];
+
+         each(args, function (value) {
+             if (value instanceof Error) {
+                 modifiedObject.err = value;
+             } else if (isObject(value)) {
+               var keys = Object.keys(value);
+                 each(keys, function (key) {
+                     modifiedObject[key] = value[key];
+                 });
+             } else {
+                 modifiedMessages.push(value);
+             }
+         });
+
+         if (modifiedObject.err) {
+             modifiedMessages.push(modifiedObject.err.message);
+         }
+         
+         modifiedArguments.push(modifiedObject);
```

Comment:
As Kate pointed out https://github.com/TryGhost/Ignition/pull/87#commitcomment-33085593, we should check if `_.isEmpty(modifiedObject)` here because we pollute the arguments with no need :wink: